src/components/common/modals/VisibleModal.tsx

```diff
@@ -11,8 +11,12 @@ export class VisibleModal extends React.Component<{ className: string, noBackgro
     }
 
     escFunction(event) {
-        if (event.keyCode === 27 && this.props.close) {
-            this.props.close(event);
+        if (event.keyCode === 27 || event.key === "Escape") {
+            if (this.props.onEscape) {
+                this.props.onEscape(event);
+            } else if (this.props.close) {
```

Comment:
Yes. This will work only when either `onEscape` or `close` prop is passed.

This was added because in the current implementation when we use the close prop, it's closing the `VisibalModal` whenever we click on the modal. It's because `close` function is called on the click event of the `VisibleModal` as well.

So didn't touch the actual implementation however provided this extra prop to be used only on the escape button event.